storage/rocksdb/ha_rocksdb.cc

```diff
@@ -9047,21 +9053,48 @@ void ha_rocksdb::get_auto_increment(ulonglong off, ulonglong inc,
   if (inc == 1 && off == 1) {
     // Optimization for the standard case where we are always simply
     // incrementing from the last position
-    *first_value = auto_incr++;
+    if ((*first_value= auto_incr) != std::numeric_limits<longlong>::max()) {
+      // Make sure we don't wrap around
+      auto_incr++;
```

Comment:
This change breaks auto-increment and two concurrent calls can end up returning the same "first_value". If you want to check before the increment, I think you need to switch this to a cas call.